src/App.jsx

```diff
@@ -0,0 +1,52 @@
+import Header from "./components/Header/Header";
+import BookList from "./containers/BookList/BookList";
+import BookModal from "./components/BookModal/BookModal";
+import styles from "./App.module.scss";
+import { useState } from "react";
+
+const App = () => {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [prevSearchTerm, setPrevSearchTerm] = useState("");
+  const [isFetchingData, setIsFetchingData] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const [searchResults, setSearchResults] = useState([]);
+  const [specificBook, setSpecificBook] = useState([]);
```

Comment:
Quite a lot of global state here. Context might help with some of the prop drilling/passing functions as props that's happening

Reply:
Research redux